WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/eclipse/logic/breakpoint/BreakpointClassifier.java

```diff
@@ -0,0 +1,38 @@
+package nl.tudelft.watchdog.eclipse.logic.breakpoint;
+
+import org.eclipse.debug.core.model.IBreakpoint;
+
+import nl.tudelft.watchdog.core.logic.breakpoint.BreakpointType;
+
+/**
+ * Estimates the nature of a breakpoint into one of {@link BreakpointType}.
+ */
+public class BreakpointClassifier {
+
+	/**
+	 * Classifies the breakpoint type of the given breakpoint by analyzing its
+	 * class name.
+	 */
+	public static BreakpointType classify(IBreakpoint breakpoint) {
+		switch (breakpoint.getClass().getSimpleName()) {
```

Comment:
Yes, that is possible using the instanceof operator and handy for possible future changes indeed. However, we then need to order these checks properly to avoid too general classifications. I'll try to indicate this clearly in the javadoc, as the ordering of the if statements now becomes important.